graph/path/bench_test.go

```diff
@@ -139,3 +139,35 @@ func BenchmarkAStarUndirectedmallWorld_100_5_20_2_Heur(b *testing.B) {
 	}
 	benchmarkAStarHeuristic(b, nswUndirected_100_5_20_2, h)
 }
+
+var (
+	gnpDirected_500_tenth   = gnpDirected(500, 0.1)
+	gnpDirected_1000_tenth  = gnpDirected(1000, 0.1)
+	gnpDirected_2000_tenth 	= gnpDirected(2000, 0.1)
+	gnpDirected_500_half    = gnpDirected(500, 0.5)
+	gnpDirected_1000_half   = gnpDirected(1000, 0.5)
+	gnpDirected_2000_half  	= gnpDirected(2000, 0.5)
+	gnpDirected_500_full    = gnpDirected(500, 1)
+	gnpDirected_1000_full   = gnpDirected(1000, 1)
+	gnpDirected_2000_full  	= gnpDirected(2000, 1)
+)
+
+func gnpDirected(n int, p float64) graph.Directed {
+	g := simple.NewDirectedGraph()
+	gen.Gnp(g, n, p, nil)
+	return g
+}
+
+func benchmarkBellmanFord(b *testing.B, n int, p float64, g graph.Directed) {
+	BellmanFordFrom(g.Node(0), g)
+}
+
+func BenchmarkBellmanFordFrom_500_tenth(b *testing.B) { benchmarkBellmanFord(b, 500, 0.5, gnpDirected_500_tenth) }
```

Comment:
These can be done with subtests rather than this way. You can either do that or I'll fix these up later. Up to you.

Reply:
I have added subtests. Are they fine?

Reply:
They're good. I should go through and fix the others.

Reply:
Thanks! If you want, I can fix the others the next days if you want. Let me know!